Source/RunActivity/Viewer3D/Processes/RenderProcess.cs

```diff
@@ -133,7 +146,12 @@ internal void Start()
             DisplaySize = new Point(GraphicsDevice.Viewport.Width, GraphicsDevice.Viewport.Height);
 
             if (Game.Settings.ShaderModel == 0)
-                Game.Settings.ShaderModel = GraphicsDevice.GraphicsDeviceCapabilities.PixelShaderVersion.Major;
+            {
+                if (GraphicsDevice.GraphicsProfile == GraphicsProfile.HiDef)
+                    Game.Settings.ShaderModel = 3;
+                else if (GraphicsDevice.GraphicsProfile == GraphicsProfile.Reach)
+                    Game.Settings.ShaderModel = 2;
```

Comment:
I'm confused by this; the shader model is set to 4.0 in the effects files so this is not right.

Reply:
I don't think this is the same. OR used the ShaderModel in a way like GraphicsProfile HiDef or Reach, but not as an exact mapping to a specific shader version. Only HiDef is used anymore with MG, and run one shader version, so either we could remove  Game.Settings.ShaderModel at all, or would need to implement correct versioning throughout the code base.

Reply:
Currently, we **are** using it to switch shader model, which is unrelated to XNA's HiDef/Reach definitions AFAIK.

This is because not everyone had shader model 3 supporting hardware/drivers when we initially introduced the newer scenery shaders. We choose based on whether version 3 is supported, and have separate version 2 and version 3 copies of the appropriate shaders (and the rest are all version 2).

For MonoGame we should remove the setting/code as everything is now shader model 4.

Reply:
wondering if we should assume DirectX11 hardware (which came out with Windows 7, 2009) and unify the shader version to 5.0 (DX11), as we wouldn't need to support pre-DX11 OS anymore